Docker packaging updates

acdha opened the pull request:
I made a few other changes in the process of setting up a test cluster. Major points:

* APT updates are always installed to ensure that any pending security updates are present
* Switch to Debian Buster as the base image to get a newer version of OpenJPEG (2.3 vs. 2.1) — this was necessary to get it to load bitonal JP2s at all and also avoids a console warning on startup about opj_decompress. https://github.com/uclouvain/openjpeg/blob/master/CHANGELOG.md suggests that this should bring some performance & memory usage improvements but I haven't measured this.

acdha committed:
Install security updates from APT

acdha committed:
Upgrade to Debian Buster

This gets OpenJPEG 2.3, avoiding a deprecation warning and getting some performance and compatibility improvements.

acdha committed:
Docker: make curl fail on errors

acdha committed:
Remove commented-out ImageMagick manual install steps

We can get these out of the Git history if they prove relevant

acdha committed:
Upgrade to Cantaloupe 4.1.2 by default

acdha committed:
Add .editorconfig and make Dockerfile indent consistently

gravesm commented:
Thanks! Given that we are using this image for our production server, we're probably going to want to do some testing with both the openjpeg and debian upgrades before merging. Unfortunately, it will likely be a few weeks before I have time.

acdha commented:
I completely understand — I'm doing the same kind of testing process here and would be happy to compare notes.

acdha committed:
Enable level 1 image info caching

acdha committed:
Enable background cache purging

acdha committed:
Remove deprecated endpoint.iiif.content_disposition feature

This has been removed in 5.0 and it causes browsers to ignore the <a download> attribute:

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-download

acdha commented:
I just pushed one other update for something we noticed — the default `endpoint.iiif.content_disposition = inline` setting sets a filename which [overrides](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-download) the value provided using `<a download="…">` in most browsers. I simply changed this to none (it's slated for removal in Cantaloupe 5) along with starting to use `?response-content-disposition=…` as part of the work which lead to https://github.com/cantaloupe-project/cantaloupe/issues/301 but I'm not sure whether that behaviour is of interest to you.
